implementations/rust/ockam/ockam_api/src/cloud/share/create.rs

```diff
@@ -0,0 +1,163 @@
+use minicbor::{Decode, Encode};
+use serde::Serialize;
+
+use super::*;
+
+#[derive(Clone, Debug, Decode, Encode, Serialize)]
+#[cbor(map)]
+#[rustfmt::skip]
+pub struct CreateInvite {
+    #[n(1)] pub expires_at: Option<String>,
+    #[n(2)] pub grant_role: RoleInShare,
+    #[n(3)] pub recipient_email: Option<String>,
+    #[n(4)] pub remaining_uses: Option<usize>,
+    #[n(5)] pub scope: ShareScope,
+    #[n(6)] pub target_id: String,
```

Comment:
What is this notion of a `target`. The unique identifier for a shared service? Maybe `shared_service_id` would be more explicit then

Reply:
The Invite mechanism on the backend is lightly polymorphic - you can also invite someone to join your Space or Project, as a feature that's independent of the Desktop app features but with a ton of conceptual overlap to a Service invite. The Service invite needs additional mandatory information to resolve, so it's special cased in a few spots. 

All that to say `target_id` is used here to be a single named argument that on the backend may be a project ID, or a space ID, and it uses `scope` argument to determine which type to attempt to resolve as.

Reply:
Ah, great I understand now. I wonder then if being a bit more specific about "ids" would be better in terms of safety:
```rust
enum ShareScope {
  SpaceScope(SpaceId),
  ProjectScope(ProjectId),
  ServiceScope(ServiceId),
}
```

Reply:
I think Eric's suggestion has good merit, so I'd like to consider it a nice-to-have and moved it to a tracking issue https://github.com/build-trust/ockam/issues/5517 to pursue later on. Most versions of this idea further decouples the client's internal data model from the backend's representation of the same ideas, so I'd like to make sure whomever tackles it does approach doing so with appropriate care.